crates/rolldown/src/module_loader/normal_module_task.rs

```diff
@@ -127,6 +142,26 @@ impl NormalModuleTask {
         let analyzed_side_effects = stmt_infos.iter().any(|stmt_info| stmt_info.side_effect);
         DeterminedSideEffects::Analyzed(analyzed_side_effects)
       });
+
+    let default_side_effects = self
```

Comment:
What is the priority between `hook_side_effects`, `packagejson` and `analyzed`, could you elaborate it?

Reply:
The `hook_side_effects` is set to the module by user manual, so the priority is high. After it should be `packagejson`, and then `analyzed`. You can see here https://github.com/rollup/rollup/blob/master/src/Module.ts#L694 and https://github.com/rollup/plugins/blob/master/packages/node-resolve/src/index.js#L254

Reply:
got it